sdk/Pulumi.Automation/Commands/IPulumiCommand.cs

```diff
@@ -5,11 +5,14 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Pulumi.Automation.Events;
+using Semver;
 
 namespace Pulumi.Automation.Commands
 {
-    internal interface IPulumiCmd
+    public interface IPulumiCommand
```

Comment:
We should consider making this an abstract class, so we can add members to it in the future without it technically being a breaking change (and renaming the file to `PulumiCommand`).

```suggestion
    public abstract class PulumiCommand
```

Reply:
Don't default interface methods makes this less of a concern now?

Reply:
Less of a concern, but it's still less flexible.